src/Platform/Microsoft.Testing.Platform/IPC/Serializers/BaseSerializer.cs

```diff
@@ -52,6 +53,29 @@ protected static void WriteString(Stream stream, string str)
         }
     }
 
+    protected static void WriteSize(Stream stream, string str)
```

Comment:
nit: I don't know if we should call it WriteStringSize() to have clearer code during the read, up to you if you feel good with that I'm fine.